web/src/app/models/game.ts

```diff
@@ -93,4 +93,48 @@ export class Game {
       const answer = new Entry({ tiles, clue });
     });
   }
+
+  private printRow(row: Tile[]): string {
+    let rowStr = '';
+    row.forEach((tile: Tile) => {
+      if (tile.value === -1) {
+        rowStr += '#';
+      } else if (tile.value === '') {
+        rowStr += ' ';
+      } else {
+        if (tile.correct) {
+          rowStr += tile.value;
+        } else {
+          rowStr += ' ';
+        }
+      }
+    });
+    return rowStr;
+  }
+
+  public serializeBoard(): string {
+    let boardStr = '';
+    this.board.forEach((row: Tile[]) => {
+      boardStr += this.printRow(row);
+    });
+    return boardStr;
+  }
+
+  public deserializeBoard(boardStr: string) {
+    if (boardStr.length !== this.crossword.height * this.crossword.width) {
+      throw Error('incorrectly formatted board string');
+    }
+
+    for (let i = 0; i < boardStr.length; i++) {
+      const str = boardStr.charAt(i);
+      if (str !== ' ' && str !== '#') {
```

Comment:
Should we set this to -1 if str === '#'

Reply:
uh i dont really agree, we send state before we receive it back. we should trust the client side state.